File: python/BlockingSemaphore.py
#!/usr/bin/env python

from Gepard import Client, Semaphore
import time

# ==========================================================================

client = Client.getInstance()

def on_error ( err, info ):
	print ( err )

client.onError ( on_error )

print ( "Acquire semaphore=user:4711" )
print ( "  blocking mode" )

sem = Semaphore ( "user:4711" )
sem.acquire()

if sem.isOwner():
	print ( sem )
	print ( "Sleep for 10 seconds" )
	time.sleep(10)
	sem.release()
	print ( "sem released." )
else:
	print ( sem )

File: java/src/org/gessinger/gepard/xmp/BlockingSemaphore.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;

public class BlockingSemaphore
{
  static public void main ( String[] args )
  {
    Util.argsToProperties ( args ) ;
    try
    {
      Semaphore sem = new Semaphore ( "user:4711" ) ;
      System.out.println ( "Trying to acquire the semaphore 'user:4711'" ) ;

      sem.acquire(5000) ;

      if ( ! sem.isOwner() )
      {
        System.out.println ( "Not owner after 5000 millis. Wait forever" ) ;
        sem.acquire() ;
      }
      System.out.println ( "Now owner of semaphore=" + sem.getName() ) ;
      System.out.println ( "sleep for 10 seconds" ) ;
      Thread.sleep ( 10000 ) ;
      System.out.println ( "release" ) ;
      sem.release() ;
      sem.getClient().close() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
}
